zboxcore/sdk/allocation_test.go

```diff
@@ -0,0 +1,3451 @@
+			expectedStr := string(expectedBytes)
+			for blobberIdx, blobber := range blobbers {
+				expectedStr = strings.ReplaceAll(expectedStr, blobberIDMask(blobberIdx+1), blobber.ID)
+				expectedStr = strings.ReplaceAll(expectedStr, blobberURLMask(blobberIdx+1), blobber.Baseurl)
+			}
+			expectedBytes = []byte(expectedStr)
+			var expected map[string]*BlobberAllocationStats
+			err := json.Unmarshal(expectedBytes, &expected)
+			assertion.NoErrorf(err, "Error json.Unmarshal() cannot parse blobber stats result format: %v", err)
+			got := a.GetBlobberStats()
+			if expected == nil || len(expected) == 0 {
+				assertion.EqualValues(expected, got)
+				return
+			}
+
+			assertion.NotEmptyf(got, "Error no blobber stats result found")
+			for key, val := range expected {
+				assertion.NotNilf(got[key], "Error result map must be contain key %v", key)
+				assertion.EqualValues(val, got[key])
+			}
+		})
+	}
+}
+
+func TestAllocation_isInitialized(t *testing.T) {
+	tests := []struct {
+		name                                        string
+		sdkInitialized, allocationInitialized, want bool
+	}{
+		{
+			"Test_Initialized",
+			true, true, true,
+		},
+		{
+			"Test_SDK_Uninitialized",
+			false, true, false,
+		},
+		{
+			"Test_Allocation_Uninitialized",
+			true, false, false,
+		},
+		{
+			"Test_Both_SDK_And_Allocation_Uninitialized",
+			false, false, false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			originalSDKInitialized := sdkInitialized
+			defer func() { sdkInitialized = originalSDKInitialized }()
+			sdkInitialized = tt.sdkInitialized
+			a := &Allocation{initialized: tt.allocationInitialized}
+			got := a.isInitialized()
+			assertion := assert.New(t)
+			if tt.want {
+				assertion.True(got, `Error a.isInitialized() should returns "true"", but got "false"`)
+				return
+			}
+			assertion.False(got, `Error a.isInitialized() should returns "false"", but got "true"`)
+		})
+	}
+}
+
+func TestAllocation_UpdateFile(t *testing.T) {
+	// setup mock miner, sharder and blobber http server
+	miner, closeMinerServer := mock.NewMinerHTTPServer(t)
+	defer closeMinerServer()
+	sharder, closeSharderServer := mock.NewSharderHTTPServer(t)
+	defer closeSharderServer()
+
+	var blobberMocks = []*mock.Blobber{}
+	var blobberNums = 4
+	for i := 0; i < blobberNums; i++ {
+		blobberIdx := mock.NewBlobberHTTPServer(t)
+		blobberMocks = append(blobberMocks, blobberIdx)
+	}
+
+	defer func() {
+		for _, blobberMock := range blobberMocks {
+			blobberMock.Close(t)
+		}
+	}()
+	// setup mock sdk
+	setupMockInitStorageSDK(t, configDir, []string{miner}, []string{sharder}, []string{})
+	// setup mock allocation
+	a := setupMockAllocation(t, allocationTestDir, blobberMocks)
+
+	var localPath = allocationTestDir + "/alloc/1.txt"
+	var status = func(t *testing.T) StatusCallback {
+		scm := &mock.StatusCallback{}
+		scm.Test(t)
+		scm.On("Started", a.ID, "/1.txt", tm.Anything, tm.Anything).Maybe()
```

Comment:
Removed StatusCallback that are not used in many cases. Just make the correct number of calls assertions whenever `StatusCallback` needed.

Reply:
I actually would prefer the tests to be more intelligent which do some assertions on the status callback. Otherwise we would not get much value from the tests. I think you are already mocking the allocations so we can implement mocking behavior when processing requests such as those on requests on `uploadChan` or `repairChan`.

Should be okay to handle it on a separate PR. But great if we leave TODO to revisit them.